diagrams/lib/transitions.dart

```diff
@@ -0,0 +1,599 @@
+    );
+    final Path axes = new Path()
+      ..moveTo(area.left, area.top) // vertical axis
+      ..lineTo(area.left, area.bottom) // origin
+      ..lineTo(area.right, area.bottom); // horizontal axis
+    canvas.drawPath(axes, _axisPaint);
+    final Offset activePoint = new FractionalOffset(
+      position,
+      1.0 - curve.transform(position),
+    ).withinRect(area);
+    // The sparkline itself.
+    final Path sparkline = new Path()..moveTo(area.left, area.bottom);
+    final double stepSize = 1.0 / (area.width * ui.window.devicePixelRatio);
+    for (double t = 0.0; t <= position; t += stepSize) {
+      final Offset point = new FractionalOffset(t, 1.0 - curve.transform(t)).withinRect(area);
+      sparkline.lineTo(point.dx, point.dy);
+    }
+    canvas.drawPath(sparkline, _sparklinePaint);
+    final Offset startPoint = new FractionalOffset(
+      position,
+      1.0 - curve.transform(position),
+    ).withinRect(area);
+    final Path graphProgress = new Path()..moveTo(startPoint.dx, startPoint.dy);
+    for (double t = position; t <= 1.0; t += stepSize) {
+      final Offset point = new FractionalOffset(t, 1.0 - curve.transform(t)).withinRect(area);
+      graphProgress.lineTo(point.dx, point.dy);
+    }
+    canvas.drawPath(graphProgress, _graphProgressPaint);
+    canvas.drawCircle(new Offset(activePoint.dx, activePoint.dy), 4.0, _positionCirclePaint);
+  }
+
+  @override
+  bool shouldRepaint(SparklinePainter oldDelegate) => curve != oldDelegate.curve || position != oldDelegate.position;
+}
+
+class Sparkline extends StatelessWidget {
+  const Sparkline({Key key, this.curve, this.position});
+
+  final Curve curve;
+  final double position;
+
+  @override
+  Widget build(BuildContext context) {
+    return new CustomPaint(painter: SparklinePainter(curve, position));
+  }
+}
+
+class DecoratedBoxTransitionDiagram extends TransitionDiagram<Decoration> {
+  const DecoratedBoxTransitionDiagram({Key key}) : super(key: key);
+
+  @override
+  String get name => 'decorated_box_transition';
+
+  @override
+  Curve get curve => Curves.decelerate;
+
+  @override
+  Animation<Decoration> buildAnimation(AnimationController controller) {
+    return _decorationTween.animate(new CurvedAnimation(
+      parent: controller,
+      curve: curve,
+    ));
+  }
+
+  static const BorderRadius _beginRadius = const BorderRadius.all(const Radius.circular(50.0));
+  static const BorderRadius _endRadius = const BorderRadius.all(const Radius.circular(0.0));
+  static final DecorationTween _decorationTween = new DecorationTween(
+    begin: new BoxDecoration(
+      borderRadius: _beginRadius,
+      color: const Color(0xffffffff),
+      boxShadow: kElevationToShadow[8],
+    ),
+    end: const BoxDecoration(
+      borderRadius: _endRadius,
+      color: const Color(0xffffffff),
+    ),
+  );
+
+  @override
+  DecoratedBoxTransitionDiagramState createState() {
+    return new DecoratedBoxTransitionDiagramState();
+  }
```

Comment:
since every state is just a build method, consider just putting the build method in these widgets and having the superclass state call it.